types/module/module.go

```diff
@@ -299,19 +303,44 @@ func (m *Manager) RegisterServices(cfg Configurator) {
 	}
 }
 
-// InitGenesis performs init genesis functionality for modules. Exactly one
-// module must return a non-empty validator set update to correctly initialize
-// the chain.
+func (m *Manager) UnmarshalGenesis(cdc codec.JSONCodec, genesisData map[string]json.RawMessage) map[string]proto.Message {
+	var wg sync.WaitGroup
+	genesisStates := map[string]proto.Message{}
+	genesisStatesMutexs := map[string]*sync.Mutex{}
```

Comment:
@alexanderbez , I've added the mutexes here, is it okay ?

Reply:
Honestly, I'd just add a single mutex to the `Manager` type.